File: tc/srm627div2/500.cpp
#include <string>
#include <map>

using namespace std;
class HappyLetterDiv2 {
public:
	char getHappyLetter(string letters) {
		map<char,int> counter;
		for (string::iterator it = letters.begin(); it != letters.end(); ++it) {
			if (counter.find(*it) == counter.end())
				counter[*it] = 1;
			else
				counter[*it]++;
		}
		char best = '.';
		int best_size = letters.size() / 2;
		for (map<char,int>::iterator i = counter.begin(); i != counter.end(); ++i) {
			if (best_size < i->second) {
				best = i->first;
				best_size = i->second;
			}
		}
		return best;
	}
};

File: tc/srm627div2/500.py

# Standard Error:
# Traceback (most recent call last):
#   File "Wrapper.py", line 186, in <module>
# NameError: name 'writeChar' is not defined

# http://apps.topcoder.com/forums/?module=Thread&threadID=824794&start=0&mc=2#1900367
# Looks like this problem is not solvable in Python...


class HappyLetterDiv2:
	def getHappyLetter(self,letters):
		counter = dict()
		for letter in letters:
			counter[letter] = counter.get(letter,0) + 1
		candidate = max(counter, key = lambda key : counter[key])
		return candidate if 2 * counter[candidate] > len(letters) else '.'
